bpfd-operator/controllers/bpfd-agent/common.go

```diff
@@ -156,17 +158,20 @@ func (r *ReconcilerCommon) removeFinalizer(ctx context.Context, o client.Object,
 }
 
 // updateStatus updates the status of a BpfProgram object if needed, returning
-// if the status was already set for the given bpfProgram, meaning reconciliation
+// false if the status was already set for the given bpfProgram, meaning reconciliation
 // may continue.
 func (r *ReconcilerCommon) updateStatus(ctx context.Context, prog *bpfdiov1alpha1.BpfProgram, cond bpfdiov1alpha1.BpfProgramConditionType) bool {
 	if prog.Status.Conditions != nil {
 		// Get most recent condition
 		recentIdx := len(prog.Status.Conditions) - 1
 
-		condition := prog.Status.Conditions[recentIdx]
-
-		if condition.Type == string(cond) {
+		// If the most recent condition is the same as input, just return.
+		if prog.Status.Conditions[recentIdx].Type == string(cond) {
 			return false
+		} else {
+			// Remove the input condition from the list if it exists (may not exist)
```

Comment:
Wait what's the point of this?  Don't we want all the conditions there? It will just be printed as a list so that users can see what happened during reconciliation.

Reply:
If the condition being updated is not the latest condition (last in the list), removing it and then it is added back below and placed at the end of the list.

Reply:
For now that's alright I guess It looks like it's time to reanalyze our use of conditions lol 

In the best practices I see 

`This collection should be treated as a map with a key of type.` when it talks about the slice of conditions 😶‍🌫️  So we're definitely doing it wrong lol I don't think we should ever be looking for the "latest" condition